libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/useShareDialogBase.ts

```diff
@@ -146,9 +158,30 @@ export const useShareDialogBase = (props: IShareDialogBaseProps): IUseShareDialo
 
     useGetAccessList({ sharedObjectRef, onSuccess: onLoadGranteesSuccess, onError });
 
+    const [isUnderStrictControl, setUnderStrictControl] = useState(isOriginallyUnderStrictControl);
+    const [isLocked, setLocked] = useState(isOriginallyLocked);
+
+    const onStrictControlChange = useCallback((isLenientUnderStrictControl: boolean) => {
+        setUnderStrictControl(!isLenientUnderStrictControl);
+    }, []);
+
+    const onLockChange = useCallback((isLocked: boolean) => {
+        setLocked(isLocked);
+    }, []);
+
     const isShareDialogDirty = useMemo(() => {
-        return granteesToDelete.length !== 0;
-    }, [granteesToDelete]);
+        return (
+            granteesToDelete.length !== 0 ||
+            isLocked !== isOriginallyLocked ||
```

Comment:
Could you wrap this in method "accessControlDirty" and simplify this condition.

Reply:
I left it as is but the code looks bit different now.